ruff_lsp/server.py

```diff
@@ -175,6 +205,214 @@ class VersionModified(NamedTuple):
 ]
 
 
+###
+# Document
+###
+
+
+def _uri_to_fs_path(uri: str) -> str:
+    """Convert a URI to a file system path."""
+    path = uris.to_fs_path(uri)
+    if path is None:
+        # `pygls` raises a `Exception` as well in `workspace.TextDocument`.
+        raise ValueError(f"Unable to convert URI to file path: {uri}")
+    return path
+
+
+@enum.unique
+class DocumentKind(enum.Enum):
+    """The kind of document."""
+
+    Text = enum.auto()
+    """A Python file."""
+
+    Notebook = enum.auto()
+    """A Notebook Document."""
+
+    Cell = enum.auto()
+    """A cell in a Notebook Document."""
```

Comment:
I had to introduce the cell specific document kind now that they're being used in multiple places.